packages/ERTP/src/paymentLedger.js

```diff
@@ -291,7 +291,22 @@ export const makePaymentLedger = (
    */
   const withdraw = (currentBalance, updatePurseBalance, amount) => {
     amount = coerce(amount);
-    const newPurseBalance = subtract(currentBalance, amount);
+    let newPurseBalance;
+    try {
+      newPurseBalance = subtract(currentBalance, amount);
+    } catch (err) {
```

Comment:
Rather than try-catch and deconstructing the resulting Error, wouldn't it be safer to test if `AmountMath.isGTE(currentBalance, amount)`? Then we can construct our own error.

Reply:
I like that! I'll try it

Reply:
Tried it, and it looks good. So it sounds like the best practice is to assert ahead of any errors that may be common?